kolibri/core/notifications/signals.py

```diff
@@ -0,0 +1,77 @@
+import uuid
+
+from django.db import transaction
+from django.db.models.signals import post_save
+from django.dispatch import receiver
+
+from .models import KolibriNotification
+from .models import NotificationType
+from kolibri.core.lessons.models import Lesson
+from kolibri.core.logger.models import ContentSummaryLog
+# from .models import HelpReason
+# from kolibri.core.exams.models import Exam
+
+
+def get_assignments(memberships, content_id):
+    """
+    Returns all Lessons assigned to the user having the content_id
+    """
+    learner_groups = [m.collection for m in memberships]
+
+    # Return only active Lessons that are assigned to the requesting user's groups
+    filtered_lessons = Lesson.objects.filter(
+        lesson_assignments__collection__in=learner_groups,
+        is_active=True,
+        resources__regex=r"" + content_id + ""
+    ).distinct()
+    lesson_resources = {lesson.collection_id: (lesson.id, lesson.resources) for lesson in filtered_lessons}
+    return lesson_resources
+
+
+@receiver(post_save, sender=ContentSummaryLog)
```

Comment:
There's an interesting perspective that [signals decrease code readability](https://lincolnloop.com/blog/django-anti-patterns-signals/), and it's equally effective to call this function directly from `ContentSummaryLog.save()`

I tend to agree because this function is already tightly coupled to the structure of `ContentSummaryLog`. I believe both strategies allow us to migrate to a background queue, but maybe I'm missing something and signals make that easier.

thoughts?

Reply:
One of my first approaches was using `ContentSummaryLog.save()`. But with that method the actual save happens after this function is executed. This has two undesired side-effects:
1. saving of `ContentSummaryLog` is delayed by the function
2. any aggregated query made by the function does not have the instance of ContentSummaryLog in the calculations

Point 2 made me discard using `ContentSummaryLog.save()`

Reply:
Done in an async thread. Chosen that option based on the work from Richard on #4225 because creating and executing a Django command everytime a logger saves is a clear overhead. The queue is set to be procesed every 5 seconds.